build_docs.pl

```diff
@@ -848,4 +849,25 @@ sub usage {
                             Specified by build_docs when running in its container
 
 USAGE
+    if ( $Opts->{in_standard_docker} ) {
+        say <<USAGE;
+    Self Test:
+
+        $name --self-test <args to pass to make>
+
+    `--self-test` is a wrapper around `make` which is used exclusively for
+    testing. Like `make`, the current directory selects the `Makefile` and
+    you can make specific targets. Some examples:
+
+    Execute all tests:
+        $name --self-test
+
+    Execute all of the tests for our extensions to Asciidoctor:
+        $name --self-test -C resources/asciidoctor
+
+    Run rubocop on our extensions to Asciidoctor:
+        $name --self-test -C resources/asciidoctor rubocop
```

Comment:
Full transparency, I am still pretty "ehhh" on two ways to do the same thing instead of standardizing on e.g. `make test` and directing users to it. Fully granting the distinction between the goal and the path we take to get there, of course.

Reply:
`make test` won't work because it isn't in the docker container so it probably won't have the dependencies available. If it did i wouldn't have made this.

Reply:
I'm not really worried about how the _tooling_ accomplishes running it (barring a case where a maintenance headache gets introduced), but I am concerned with us telling users (and this is the usage string) to do this kind of stuff. It seems nicer if users always go to the same place with the same expectations. For example, they should be able to type `make rubocop` from the top level, and have that do the expected thing they wanted. 

> `make test` won't work because it isn't in the docker container so it probably won't have the dependencies available.

I don't understand the problem. If something is missing from the container, add it to the container? At the same time I also am unsure how the container is necessarily involved here, if this is a wrapper around make then it's running make, and if that isn't what's going on then this usage string is inaccurate, because that's what it says it does.

Additionally, speaking as something a user would do, a user would check out the repo, cd into it, and type `make test`. That's on their host; running any containers as a result of that is directly within our control and also only marginally their problem -- like, they need to have docker, but that's about it. That's what I'm really getting at in this comment thread: a simplified, consistent, and unsurprising experience for contributors. 

I imagine that over time they will become completely insulated from `build_docs.pl` ever talking to them at all, at least if I understand the current plan? Which ultimately ends in `build_docs.pl` no longer existing at all?

Reply:
Right now `build_docs` runs docker which runs make to execute the tests. That is the only way to run these tests. This way the Makefiles get to "know" that they are in our standard docker container. The idea is that the only thing that needs to be portable is `build_docs`.

With this change the user doesn't have to know about `build_docs.pl`. I rename it to `build_docs` in the usage instructions if you are running in docker.  They do still have to know about `build_docs` which is weird. Folks are going to be used to make. But I *like* that I know exactly the shape of the environment that the makefile is executing within. 

If there was a way I could hook into make's startup to transparently restart itself in the docker container that'd be lovely.

Reply:
I'm sure there's some way around it. I'm not _so_ worried that it can't be on a future PR, though. Let's think about it more, and I will stop expecting anything _here_ about it. :)